tests/functional/ctu_failure/test_ctu_failure.py

```diff
@@ -0,0 +1,151 @@
+# -----------------------------------------------------------------------------
+""" CTU function test."""
+
+import json
+import os
+import shutil
+import subprocess
+import unittest
+import zipfile
+
+from libtest import env
+
+NO_CTU_MESSAGE = "CTU is not supported"
+
+
+class TestCtu(unittest.TestCase):
+    """ Test CTU functionality. """
+
+    def setUp(self):
+        """ Set up workspace."""
+
+        # TEST_WORKSPACE is automatically set by test package __init__.py .
+        self.test_workspace = os.environ['TEST_WORKSPACE']
+
+        test_class = self.__class__.__name__
+        print('Running ' + test_class + ' tests in ' + self.test_workspace)
+
+        # Get the CodeChecker cmd if needed for the tests.
+        self._codechecker_cmd = env.codechecker_cmd()
+        self.env = env.codechecker_env()
+        self.report_dir = os.path.join(self.test_workspace, 'reports')
+        os.makedirs(self.report_dir)
+        self.test_dir = os.path.join(os.path.dirname(__file__), 'test_files')
+
+        # Get if clang is CTU-capable or not.
+        cmd = [self._codechecker_cmd, 'analyze', '-h']
+        output = subprocess.check_output(cmd, stderr=subprocess.STDOUT,
+                                         cwd=self.test_dir, env=self.env)
+        self.ctu_capable = '--ctu-' in output
+        print("'analyze' reported CTU-compatibility? " + str(self.ctu_capable))
+
+        # Fix the "template" build JSONs to contain a proper directory
+        # so the tests work.
+        raw_buildlog = os.path.join(self.test_dir, 'buildlog.json')
+        with open(raw_buildlog) as log_file:
+            build_json = json.load(log_file)
+            for command in build_json:
+                command['directory'] = self.test_dir
+
+        self.__old_pwd = os.getcwd()
+        os.chdir(self.test_workspace)
+        self.buildlog = os.path.join(self.test_workspace, 'buildlog.json')
+        with open(self.buildlog, 'w') as log_file:
+            json.dump(build_json, log_file)
+
+    def tearDown(self):
+        """ Tear down workspace."""
+
+        os.chdir(self.__old_pwd)
+        shutil.rmtree(self.report_dir, ignore_errors=True)
+
+    def test_ctu_logs_ast_import(self):
+        """ Test that Clang indeed logs the AST import events.
+        """
+        if not self.ctu_capable:
+            self.skipTest(NO_CTU_MESSAGE)
+        output = self.__do_ctu_all(reparse=False,
+                                   extra_args=["--verbose", "debug"])
+        self.assertIn("ANALYZE (CTU loaded AST for source file)", output)
+
+    def test_ctu_failure_zip(self):
+        """ Test the failure zip contains the source of imported TU"
+        """
+
+        if not self.ctu_capable:
+            self.skipTest(NO_CTU_MESSAGE)
+
+        output = self.__do_ctu_all(reparse=False,
+                                   extra_args=[
+                                       "--verbose", "debug",
+                                       "-e", "debug.ExprInspection"
```

Comment:
Please add a comment why do we enable/use this checker for this test.

Reply:
Ok , I added a comment.